### PR TITLE
cxx-core: add segment property to properties class

### PR DESCRIPTION
**Pull request by r-rojo**

### CODE REVIEW

File: libopaecxx/src/properties.cpp [hunk 1]
```diff
@@ -38,6 +38,7 @@ properties::properties()
     : props_(nullptr),
       type(&props_, fpgaPropertiesGetObjectType, fpgaPropertiesSetObjectType),
       num_errors(&props_, fpgaPropertiesGetNumErrors, fpgaPropertiesSetNumErrors),
+      segment(&props_, fpgaPropertiesGetSegment, fpgaPropertiesSetSegment),
```
Comment (asgibbons) [inline libopaecxx/src/properties.cpp]: I don't see where these methods are defined.  Is that a different commit?

File: tests/function/gtCxxProperties.cpp [hunk 2]
```diff
@@ -144,3 +144,16 @@ TEST(LibopaecppPropsCommonALL, get_num_errors) {
...
+ * @test segment
+ * Given a properties properties object with the segment property set to a
+ * known value
+ * When I get the segment property
+ * Then the number is the expected value
+ */
+TEST(LibopaecppPropsCommonALL, get_segment) {
+  auto p = properties::get();
+  p->segment = 9090;
+  EXPECT_EQ(static_cast<uint16_t>(p->segment), 9090);
```
Comment (asgibbons) [inline tests/function/gtCxxProperties.cpp]: So, you're assigning a value to a variable, then expecting it to be equal to the value you just assigned to it?  I don't see where this is actually testing anything, but there may be some C++ magic here somewhere.  Can you clarify?

File: libopaecxx/src/properties.cpp (hunk 1 above)
Comment (r-rojo) [reply libopaecxx/src/properties.cpp]: which methods? fpgaProperties[GS]etSegment?

File: tests/function/gtCxxProperties.cpp (hunk 2 above)
Comment (r-rojo) [reply tests/function/gtCxxProperties.cpp]: line 157 is exercising the assignment operator in the pvalue template class which will ultimately call fpgaPropertiesSetSegment.
line 158 is exercising the conversion operator in the pvalue template classs which will call fpgaPropertiesGetSegment to retrieve the value.

**Pull request merged**